contrib/enfuseAdvanced.lua

```diff
@@ -54,548 +54,146 @@ Same as other export modules
-        enf_cmbx_contrast_edge_scale.value=dt.preferences.read("enfuseAdvanced", pref_text.."contrast_edge_scale", "integer")
-        enf_cmbx_contrast_min_curvature.value=dt.preferences.read("enfuseAdvanced", pref_text.."contrast_min_curvature", "integer") 
-        enf_sldr_exposure_weight.value=dt.preferences.read("enfuseAdvanced", pref_text.."exposure_weight", "float")
-        enf_sldr_saturation_weight.value=dt.preferences.read("enfuseAdvanced", pref_text.."saturation_weight", "float")
-        enf_sldr_contrast_weight.value=dt.preferences.read("enfuseAdvanced", pref_text.."contrast_weight", "float")
-        enf_sldr_exposure_optimum_weight.value=dt.preferences.read("enfuseAdvanced", pref_text.."exposure_optimum_weight", "float")
-        enf_sldr_exposure_width_weight.value=dt.preferences.read("enfuseAdvanced", pref_text.."exposure_width_weight", "float")   
-	
-    end,
-    "1 - DRI image", "2 - DRI image","3 - DRI image","1 - DFF image", "2 - DFF image","3 - DFF image","free preset",  
-    
-    reset_callback = function(self_type)
-       enf_cmbx_fusion_type.value = dt.preferences.read("enfuseAdvanced", "def_fusion_type", "integer")
-    end
-} 
-enf_entry_tag = dt.new_widget("entry"){
-	tooltip = "Additional tags to be added on import. Seperate with commas, all spaces will be removed",
-	text = pref_addtags,
-	placeholder = "Enter tags, seperated by commas",
-	editable = true
-}
-enf_widget = dt.new_widget("box") {
-    orientation = "vertical",
-    enf_label_align_options,
-    enf_chkbtn_radial_distortion,
-    enf_chkbtn_optimize_field,
-    enf_chkbtn_optimize_image_center,
-    enf_chkbtn_auto_crop,
-    enf_chkbtn_distortion,
-    enf_cmbx_grid_size,
-    enf_cmbx_control_points,
-    enf_cmbx_control_points_remove,
-    enf_cmbx_correlation,
-    enf_separator1,
-    enf_separator1b,
-    enf_label_enfuse_options,
-    enf_cmbx_fusion_type,
-    enf_chkbtn_image_variations,
-    enf_sldr_exposure_weight,
-    enf_sldr_saturation_weight,
-    enf_sldr_contrast_weight,
-    enf_sldr_exposure_optimum_weight,
-    enf_sldr_exposure_width_weight, 
-    enf_chkbtn_hard_masks,
-    enf_chkbtn_save_masks,
-    enf_cmbx_contrast_edge_scale,
-    enf_cmbx_contrast_min_curvature,
-    enf_cmbx_contrast_window_size,
-    enf_button_save_preset,
-    enf_button_load_preset,
-    enf_separator2,
-    enf_separator2b,
-    enf_label_output_format,
-    enf_cmbx_output_format,
-    enf_label_path,
-    enf_file_chooser_button_path,
-	enf_chkbtn_source_location,
-    enf_cmbx_existing_file,
-	enf_cmbx_style,
-	enf_chkbtn_cpytags,
-	enf_entry_tag,
-}
+end
 
--- FUNCTION -- 
-local function GetFileName(full_path)
-	--[[Parses a full path (path/filename_identifier.extension) into individual parts
-	Input: Folder1/Folder2/Folder3/Img_0001.CR2
+local function GetFileName(full_path) -- Parses a full path (path/filename_identifier.extension) into individual parts
```

Comment:
There's already path parsing functions in the file library including split_filepath which returns path, filename, basename, and extension

Reply:
I thought I looked at incorporating that the first time 'round and it didn't quite fit my needs. I could be wrong. I'll look again.

Reply:
I looked again and split_filepath  doesn't quite do the samething as what I do here. I could use it, but then I'd have to do further processing on the result, so I would still have to have a separate function to return exactly what I'm looking for (I return the strictly numerical part of the image name as the 'ID'. Here you can see what is returned from each and how they differ:

--Split Filename--
C:\Users\ERTELK\AppData\Local\Temp\
_MG_4839.tif
_MG_4839
tif

--Get Filename--
C:\Users\ERTELK\AppData\Local\Temp\
_MG_4839
4839
.tif